src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,24 +1611,1318 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+            {
+                case DISPLAY_MALE_01_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_MALE_01_F;
+                    break;
+                case DISPLAY_MALE_02_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_MALE_02_F;
+                    break;
+                case DISPLAY_MALE_03_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_MALE_03_F;
+                    break;
+                case DISPLAY_MALE_04_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_MALE_04_F;
+                    break;
+                case DISPLAY_FEMALE_01_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_FEMALE_01_F;
+                    break;
+                case DISPLAY_FEMALE_02_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_FEMALE_02_F;
+                    break;
+                case DISPLAY_FEMALE_03_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_FEMALE_03_F;
+                    break;
+                case DISPLAY_FEMALE_04_F:
+                    spellId = SPELL_FORSAKEN_TROOPER_FEMALE_04_F;
+                    break;
+                default:
+                    break;
+            }
+
+            hillsbradRefugee->CastSpell(hillsbradRefugee, spellId, true);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_forsaken_trooper_masterscript_fenris_isle::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+enum HillsbradRefugee
+{
+    QUEST_RISE_FORSAKEN                         = 27097,
+
+    SPELL_TROOPER_MASTERSCRIPT_FENRIS           = 83997,
+    SPELL_NOTIFY_AGATHA_FENRIS                  = 83990,
+    SPELL_LORDAERON_MIGHT                       = 87104,
+
+    EVENT_LORDAERON_MIGHT                       = 1
+};
+
+// 44954, 44966 - Hillsbrad Refugee
+struct npc_silverpine_hillsbrad_refugee : public ScriptedAI
+{
+    npc_silverpine_hillsbrad_refugee(Creature* creature) : ScriptedAI(creature) {}
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        switch (spellInfo->Id)
+        {
+            case SPELL_RISE_FORSAKEN_FENRIS:
+                DoCastSelf(SPELL_TROOPER_MASTERSCRIPT_FENRIS);
+                me->DespawnOrUnsummon(2s);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        if (Player* player = killer->ToPlayer())
+        {
+            if (player->GetQuestStatus(QUEST_RISE_FORSAKEN) == QUEST_STATUS_INCOMPLETE)
+                me->CastSpell(player, SPELL_NOTIFY_AGATHA_FENRIS, true);
+        }
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_LORDAERON_MIGHT, 3s, 8s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
```

Comment:
move below `if (!UpdateVictim())` as its only related to infight stuff

Reply:
Feedback applied.